base_custom_attributes/custom_attributes.py

```diff
@@ -0,0 +1,449 @@
+# -*- encoding: utf-8 -*-
+###############################################################################
+#                                                                             #
+#   base_attribute.attributes for OpenERP                                     #
+#   Copyright (C) 2011 Akretion Benoît GUILLOT <benoit.guillot@akretion.com>
+#   Copyright (C) 2013 Akretion Raphaël VALYI <raphael.valyi@akretion.com>
+#                                                                             #
+#   This program is free software: you can redistribute it and/or modify      #
+#   it under the terms of the GNU Affero General Public License as            #
+#   published by the Free Software Foundation, either version 3 of the        #
+#   License, or (at your option) any later version.                           #
+#                                                                             #
+#   This program is distributed in the hope that it will be useful,           #
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of            #
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the             #
+#   GNU Affero General Public License for more details.                       #
+#                                                                             #
+#   You should have received a copy of the GNU Affero General Public License  #
+#   along with this program.  If not, see <http://www.gnu.org/licenses/>.     #
+#                                                                             #
+###############################################################################
+
+import ast
+from openerp import models, fields, api
+from openerp.osv import orm
+from openerp.tools.translate import _
+from lxml import etree
+from unidecode import unidecode  # Debian package python-unidecode
+import re
+
+
+def safe_column_name(string):
+    """This function prevent portability problem in database column name
+    with other DBMS system
+    Use case : if you synchronise attributes with other applications """
+    string = unidecode(string.replace(' ', '_').lower())
+    return re.sub(r'[^0-9a-z_]', '', string)
+
+
+class AttributeOption(models.Model):
+    _name = "attribute.option"
+    _description = "Attribute Option"
+    _order = "sequence"
+
+    @api.model
+    def _get_model_list(self):
+        model_pool = self.env['ir.model']
+        models = model_pool.search([])
+        return [(model.model, model.name) for model in models]
+
+    name = fields.Char(
+        size=128,
+        translate=True,
+        required=True)
+    value_ref = fields.Reference(
+        string='Reference',
+        selection='_get_model_list',
+        size=128)
```

Comment:
You can surely remove that size limit as you get value from `_get_model_list` unless I'm missing something